txn_writer.go

```diff
@@ -294,6 +294,14 @@ func (sw *SafeWriter) Write(root string, sm gps.SourceManager) error {
 		}
 	}
 
+	// Ensure vendor/.git is preserved if present
+	if hasDotGit(vpath) {
+		err = os.Rename(filepath.Join(vpath, ".git"), filepath.Join(td, "vendor/.git"))
```

Comment:
`dep.renameWithFallback` does what you have here, plus a bunch more error handling, let's use that instead of `os.Rename` directly.